api/v1/overlaynetwork_types.go

```diff
@@ -20,34 +20,47 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // OverlayNetworkSpec defines the desired state of OverlayNetwork
 type OverlayNetworkSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Name is the NetworkAttachmentDefinition name.
+	Name string `json:"name"`
 
-	// Foo is an example field of OverlayNetwork. Edit overlaynetwork_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
-}
+	// NetworkName is the name of the overlay network.
+	// The following name formats are supported:
+	// - <namespace>/<name> : Used when cross namespace connectivity is desired.
+	// - <name> : The name will be prefixed with the current namespace to assure uniqueness across namespaces.
+	NetworkName string `json:"networkName"`
+
+	// The maximum transmission unit (MTU).
+	// The default value, 1300, is automatically set by the kernel.
+	// +optional
+	Mtu string `json:"mtu,omitempty"`
+
+	// The subnet to use for the network across the cluster.
+	// Only include the CIDR for the node. E.g. 10.100.200.0/24.
+	//
+	// IPv6 (2001:DBB::/64) and dual-stack (192.168.100.0/24,2001:DBB::/64) subnets are supported.
+	//
+	// When omitted, the logical switch implementing the network only provides layer 2 communication,
+	// and users must configure IP addresses for the pods.
+	// Port security only prevents MAC spoofing.
+	// +optional
+	Subnets string `json:"subnets,omitempty"`
```

Comment:
Is it legit with overlay to omit the Subnets?

Reply:
> Do overlay networks support l2?

Yes. That's pretty much the [way to configure](https://docs.openshift.com/container-platform/4.15/virt/vm_networking/virt-connecting-vm-to-ovn-secondary-network.html#virt-creating-layer2-nad-cli_virt-connecting-vm-to-ovn-secondary-network) the network for virt use cases downstream.